ensure_tiller_installed.go

```diff
@@ -261,10 +261,10 @@ func (c *Client) EnsureTillerInstalledWithValues(ctx context.Context, values []s
 
 		o := func() error {
 			t, err := c.newTunnel()
-			if !installTiller && IsTillerNotFound(err) {
-				return backoff.Permanent(microerror.Mask(err))
-			} else if err != nil {
+			if IsTillerNotFound(err) || IsTooManyResults(err) {
 				return microerror.Mask(err)
+			} else if err != nil {
+				return backoff.Permanent(microerror.Mask(err))
```

Comment:
AIUI the else is now a permanent backoff? But can there be cases we've missed?

If so we may not retry in situations where we currently do. That would be bad if we require another reconciliation loop to install tiller in tenant clusters.

Reply:
@rossf7 By checking 3 all possible outcomes from tiller deployment phase (Tiller not found, multiple Tiller pods, version not matching), I think these are enough. 
And about stopping reconciliation of this  `backoff.Permanent` error cases, I think that depends on the behavior of caller, right? 
Do you think to make sense have another error, something like `ErrorStartingiTiller` as mask, and return in this case, so we can cancel our reconciliation?

Reply:
The approach makes sense. I'm fine with this once e2e is green and it has been tested manually.